dot/core/service.go

```diff
@@ -526,7 +526,7 @@ func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
 	responseData := []*types.BlockData{}
 
 	for _, hash := range subchain {
-		data, err := s.blockState.GetBlockData(hash)
+		data, err := s.blockState.GetBlockDataAllFields(hash)
```

Comment:
looking at the rest of the PR, if you want to move the bitmap parsing to a function in the state that would be fine with me. it would help clean up the state interface also.

for example:
```
func (b *BlockState) GetBlockData(bitmap byte) (*types.BlockData) {
    ret = new(types.BlockData)
    if (bitmap&1) == 1 // get header and set ret.Header
    if (bitmap&2)>>1 == 1 // get body and set ret.Body
    // etc for rest of fields
}
```